### ISSUE TITLE
A couple of suggestions

### ISSUE THREAD

**Issue by kdoshere** [Bionus/imgbrd-grabber#1005]
Hi,

I was using the app and looking around and noticed that when you browse in any page, if you right click over an image you get the save and delete options independently of the value of the "folder" field. But when you see the same image using the image viewer it won't show if that image exists or not unless the "folder" field is pointing to the folder in which the image is located.

I suppose that when right clicking the image it takes the data from the md5 list.txt while when using the visor it takes it from the directory, is this correct?

Would it be possible to modify the visor so it could do the same? I know it is a bit heavy when you have to deal with a number of folders/sub-folders, having to select each one of the (new) images to see if you already downloaded it or not (I'm kind of clumsy sometimes and I forget if I already did it or not).

The next thing is the MD5 list fixer

I'm using [%date:format=yyyy.MM.dd - hh.mm.ss%] %character% (%copyright%) drawn by %artist% %md5%.%ext% as filename and when downloading the images it will sort them correctly in the md5 list.txt but because of a mistake the list was lost so when I tried to regenerate it I realized that the "Get MD5 in filename" option wasn't able to read the MD5 of the files even if that information was in the filename. So in that case I can only use the "Force MD5 calculation" option, it worked alright but this can be extremely slow (when doing batches of 200k+ files).

So I was wondering if this could be done in another way because if we use more "exotic" filenames other than %md5%.%ext% there's no other way than using the force MD5 calculation.

Thanks for your attention and keep up the good work.

**Comment by Bionus** [comment]
> I was using the app and looking around and noticed that when you browse in any page, if you right click over an image you get the save and delete options independently of the value of the "folder" field. But when you see the same image using the image viewer it won't show if that image exists or not unless the "folder" field is pointing to the folder in which the image is located.
>
> I suppose that when right clicking the image it takes the data from the md5 list.txt while when using the visor it takes it from the directory, is this correct?
>
> Would it be possible to modify the visor so it could do the same? I know it is a bit heavy when you have to deal with a number of folders/sub-folders, having to select each one of the (new) images to see if you already downloaded it or not (I'm kind of clumsy sometimes and I forget if I already did it or not).

They already have the same "saving" behaviour, both take into account the MD5 list.

The difference being that in the image window, because of some wrong UX choices in a previous update, the button is still called "Save" and will show a "Ignored!" message if you click it (while in the context menu it will show a "Delete" link). See issue #865 for more information on that.

> I'm using [%date:format=yyyy.MM.dd - hh.mm.ss%] %character% (%copyright%) drawn by %artist% %md5%.%ext% as filename and when downloading the images it will sort them correctly in the md5 list.txt but because of a mistake the list was lost so when I tried to regenerate it I realized that the "Get MD5 in filename" option wasn't able to read the MD5 of the files even if that information was in the filename. So in that case I can only use the "Force MD5 calculation" option, it worked alright but this can be extremely slow (when doing batches of 200k+ files).
>
> So I was wondering if this could be done in another way because if we use more "exotic" filenames other than %md5%.%ext% there's no other way than using the force MD5 calculation.

You should already be able to do that. More complexes filenames should be supported, as they are actually translated into a regular expression.

The basic `%md5%.%ext%` would give `(\w{32})\..*`. In your case, the result would be something like `\[.*\] .* \(.*\) drawn by .* (\w{32})\..*`.

Then, all files matching this pattern will be listed and the MD5 extracted directly from the filename.

If it doesn't, then it's a bug :smile:
I'll take a look and see if I can reproduce your problem.

**Comment by kdoshere** [comment]
Hi,

> They already have the same "saving" behaviour, both take into account the MD5 list.
> 
> The difference being that in the image window, because of some wrong UX choices in a previous update, the button is still called "Save" and will show a "Ignored!" message if you click it (while in the context menu it will show a "Delete" link). See issue #865 for more information on that.

I see, it seems like I should have searched a little bit more, anyways its good to see what was the issue.

I was looking why the "Get MD5 in filename" option didn't work, after a while I tried to fiddle around a little with the filenames and I realized that:

[%date:format=yyyy.MM.dd - hh.mm.ss%] %character% (%copyright%) drawn by %artist% %md5%.%ext%
[2017.01.05 - 21.19.53] heles (granblue fantasy) drawn by ao (time-leap) 2ef89a01ab2981098c1ae9e4f155fadc.png

Doesn't create an entry on the md5.txt file, but if I use:

%character% (%copyright%) drawn by %artist% %md5%.%ext%
heles (granblue fantasy) drawn by ao (time-leap) 2ef89a01ab2981098c1ae9e4f155fadc.png

It will successfully create the entry on the md5.txt file. So after trying some other ways without the "[]" it seems that the issue are the "." and if we replace them with "-":

[%date:format=yyyy-MM-dd - hh-mm-ss%] %character% (%copyright%) drawn by %artist% %md5%.%ext%
[2017-01-05 - 21-19-53] heles (granblue fantasy) drawn by ao (time-leap) 2ef89a01ab2981098c1ae9e4f155fadc.png

If we do that the "Get MD5 in filename" works correctly and it will create the output as expected. So, with that in mind we can conclude that Grabber doesn't like the dot in the date:format.

Well that was just my observation and this could be considered as a minnor issue in the matter, by the way I'm using Windows 10 if that matters.
